src/api/middleware/validators/offer.js

```diff
@@ -452,10 +454,30 @@ export const setDefaultValuesCreate = (req, res, next) => {
 };
 
 export const get = useExpressValidators([
-    query("offset", ValidationReasons.DEFAULT)
+    query("lastOfferId", ValidationReasons.DEFAULT)
         .optional()
-        .isInt({ min: 0 }).withMessage(ValidationReasons.INT)
-        .toInt(),
+        .custom(isObjectId).withMessage(ValidationReasons.OBJECT_ID).bail()
+        .custom(existingOfferId)
+        .custom(async (offerId, { req }) => {
+            try {
+                const offerService = new OfferService();
+                const { value, ...filters } = req.query;
+
+                if (!await offerService.doesOfferMatchFilters(offerId, filters)) {
+                    throw new Error(ValidationReasons.OFFER_NOT_MATCHING_CRITERIA);
+                }
+
+                if (!value) return true;
+
+                const score = await offerService.getTextSearchScoreById(offerId, value);
+                if (score <= 0) throw new Error(ValidationReasons.OFFER_NOT_MATCHING_CRITERIA);
```

Comment:
Why are we throwing when something does not match?

Say we searched for the first batch of offers. The last one was modified, and no longer matches the filters. We won't be able to show more results in frontend because of this, which shouldn't happen. Why does the filter matching impact the next search? Even if some result no longer matches the filtering, that's not a problem as it was returned as part of a previous state. We shouldn't worry about it.

Reply:
I did this to validate a poorly made request. If the offer does not match the full text search, we won't get a positive score and we can't get more offers. In the case the offer is edited, we wouldn't get more offers even if the we remove the validator, right?

Reply:
I thought it'd be best for the FE to know what happened instead of just not receiving anything else

Reply:
I understand. But still, I think we can optimize a bit: let's send an encoded string (our `queryToken`) that contains information about both the last offer's id and the score. This way we don't have to query the db every time just to get the score. It also prevents the problem I described: if we know the score threshold and the last id, we can continue the search even if that last offer no longer exists or is not part of the results anymore.

So yes, we sen the score to the frontend, but it doesn't know about it, it's just a random string that you just send back to continue fetching stuff. The queryToken could literally be "this is the id: askdsald, and this is the last score: score" and the frontend wouldn't care, it would just send it back the next time.

For encoding/decoding, it's probably simple enough to use base64 of a stringified json with id and score, but feel free to think about other (possibly smaller) options

Reply:
@imnotteixeira I just pushed the new changes